Expensify/App
# Remove didMount to stop auto focus on PasswordForm

### Details
Hidden password element was pulling focus, and accepting any text input prior to focusing on login form (email). Added a check to see if the component is visible before pulling focus on mount, added same to the login form as per maintainer request on main issue.

### Fixed Issues

$ https://github.com/Expensify/App/issues/7199

$ https://github.com/Expensify/App/issues/7133

$ https://github.com/Expensify/App/issues/7058

$ https://github.com/Expensify/App/issues/7090

### Tests

Tested on Web, iOS, and Android
1. Make sure app is on sign on page with the "phone or email" input and not password input.
2. Check that on initial load focus is on login text input. If not type characters on keyboard. (If it is pull focus and type in characters on keyboard)
3. Submit and the password component will display. 
4. Ensure the password component does not have the characters you typed in on the previous screen when focus was not on the login component.

- [ ✓] Verify that no errors appear in the JS console

### QA Steps

1. Make sure app is on sign on page with the "phone or email" input and not password input.
2. Check that on initial load focus is on login text input. If not type characters on keyboard. (If it is pull focus and type in characters on keyboard)
3. Submit and the password component will display. 
4. Ensure the password component does not have the characters you typed in on the previous screen when focus was not on the login component.

- [ ✓] Verify that no errors appear in the JS console

### Tested On

- [ ✓] Web
- [ ✓] Mobile Web
- [✓ ] Desktop
- [ ✓] iOS
- [ ✓] Android

### Screenshots


#### Web

https://user-images.githubusercontent.com/67085884/148615986-08eb355e-b777-488b-a472-a4185e56e9e1.mp4

#### Mobile Web

https://user-images.githubusercontent.com/67085884/148616031-553e2d89-4502-49d0-a7ff-cb6b6f246ef0.mp4

#### iOS
https://user-images.githubusercontent.com/67085884/148615582-923ecb5e-df78-40e9-8d2f-cba70167ac86.mp4

#### Android

https://user-images.githubusercontent.com/67085884/148615974-742fa3a2-1a28-4756-9faf-b062e26dbe4c.mp4


#### Desktop

https://user-images.githubusercontent.com/67085884/148616718-e556c3fc-07bb-4173-9494-ace3ae3b6ba2.mov
